meta-client/src/main/java/org/apache/helix/metaclient/recipes/lock/LockInfo.java

```diff
@@ -0,0 +1,193 @@
+ */
+public class LockInfo {
+
+  // Default values for each attribute if there are no current values set by user
+  public static final String DEFAULT_LOCK_ID_TEXT = "";
+  public static final String DEFAULT_OWNER_ID_TEXT = "";
+  public static final String DEFAULT_CLIENT_ID_TEXT = "";
+  public static final String DEFAULT_CLIENT_DATA = "";
+  public static final long DEFAULT_GRANTED_AT_LONG = -1L;
+  public static final long DEFAULT_LAST_RENEWED_AT_LONG = -1L;
+  public static final Duration DEFAULT_TIMEOUT_DURATION = Duration.ofMillis(-1L);
+  private static final String ZNODE_ID = "LOCK";
+  private final ZNRecord _record;
+
+  /**
+   * The keys to lock information
+   */
+  public enum LockInfoAttribute {
+    LOCK_ID,
+    OWNER_ID,
+    CLIENT_ID,
+    CLIENT_DATA,
+    GRANTED_AT,
+    LAST_RENEWED_AT,
+    TIMEOUT
+  }
+
+  /**
+   * Initialize a default LockInfo instance
+   */
+  private LockInfo() {
+    _record = new ZNRecord(ZNODE_ID);
+    setLockInfoFields(DEFAULT_LOCK_ID_TEXT, DEFAULT_OWNER_ID_TEXT, DEFAULT_CLIENT_ID_TEXT, DEFAULT_CLIENT_DATA, DEFAULT_GRANTED_AT_LONG,
+        DEFAULT_LAST_RENEWED_AT_LONG, DEFAULT_TIMEOUT_DURATION);
+  }
+
+  /**
+   * Initialize a LockInfo with a ZNRecord, set all info fields to default data
+   * @param znRecord The ZNRecord contains lock node data that used to initialize the LockInfo
+   */
+  public LockInfo(ZNRecord znRecord) {
+    this();
+    if (znRecord != null) {
+      String lockId = znRecord.getSimpleField(LockInfoAttribute.LOCK_ID.name());
+      String ownerId = znRecord.getSimpleField(LockInfoAttribute.OWNER_ID.name());
+      String clientId = znRecord.getSimpleField(LockInfoAttribute.CLIENT_ID.name());
+      String clientData = znRecord.getSimpleField(LockInfoAttribute.CLIENT_DATA.name());
+      long grantTime = znRecord.getLongField(LockInfoAttribute.GRANTED_AT.name(), DEFAULT_GRANTED_AT_LONG);
+      long lastRenewalTime =
+          znRecord.getLongField(LockInfoAttribute.LAST_RENEWED_AT.name(), DEFAULT_LAST_RENEWED_AT_LONG);
+      long timeout = znRecord.getLongField(LockInfoAttribute.TIMEOUT.name(), DEFAULT_TIMEOUT_DURATION.toMillis());
+      Duration duration = Duration.of(timeout, ChronoUnit.MILLIS);
+      setLockInfoFields(lockId,ownerId, clientId, clientData, grantTime,
+          lastRenewalTime, duration);
+    }
+  }
+
+  /**
+   * Initialize a LockInfo with data for each field, set all null info fields to default data
+   * @param lockId value of LOCK_ID attribute
+   * @param ownerId value of OWNER_ID attribute
+   * @param clientId value of CLIENT_ID attribute
+   * @param clientData value of CLIENT_DATA attribute
+   * @param grantTime the time the lock was granted
+   * @param lastRenewalTime the last time the lock was renewed
+   * @param timeout value of TIMEOUT attribute
+   */
+  public LockInfo(String lockId, String ownerId, String clientId,
+                  String clientData, long grantTime, long lastRenewalTime, Duration timeout) {
```

Comment:
In the original design we want to take clientData from client as a generic type, then we can serialize it and store it. I'm okay with changing the design to allow user to provide a serialized string of clientData, so they are responsible of serializing / deserializing the data. But please do an analysis on more known use cases, and see if will cause a huge overhead on using lock client.

Reply:
Thank you for that. After an offline discussion with @xyuanlu , we came to the conclusion that passing in the serialized version is easier to implement. Since the user can have their data as different objects, each can serialize their object however they wish. Furthermore, considering the znRecord does not have an explicit parameter for generic objects but rather we would put it under payload, it makes more sense to use their built in `getSimpleField()`. Let me know if this makes sense and answers your concerns, thanks!

Reply:
Sure sounds good. Please make sure this is clear to the users through Javadoc.